gomock/matchers.go

```diff
@@ -207,6 +207,66 @@ func (m lenMatcher) String() string {
 	return fmt.Sprintf("has length %d", m.i)
 }
 
+type inAnyOrderMatcher struct {
+	x interface{}
+}
+
+func (m inAnyOrderMatcher) Matches(x interface{}) bool {
+	given, ok := m.prepareValue(x)
+	if !ok {
+		return false
+	}
+	wanted, ok := m.prepareValue(m.x)
+	if !ok {
+		return false
+	}
+
+	usedFromGiven := make([]bool, given.Len())
```

Comment:
Maybe we should fail fast here if lens don't match, they should be exact right?

Reply:
Good point, added early return for this case.